junebug/plugins/nginx/plugin.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import subprocess
+from os import path, remove
+from distutils.dir_util import mkpath
+from urlparse import urljoin
+
+from confmodel import Config
+from confmodel.fields import ConfigText
+
+from junebug.plugin import JunebugPlugin
+from junebug.utils import channel_public_http_properties
+
+
+log = logging.getLogger(__name__)
+DIRNAME = path.dirname(__file__)
+
+
+class NginxPluginConfig(Config):
+    '''Config for :class:`NginxJunebugPlugin`'''
+
+    vhost_file = ConfigText(
+        "The file to write the junebug nginx vhost file to",
+        default='/etc/nginx/sites-enabled/junebug.conf', static=True)
+
+    locations_dir = ConfigText(
+        "The directory to write location block config files to",
+        default='/etc/nginx/includes/junebug/', static=True)
+
+    server_name = ConfigText(
+        "Server name to use for nginx vhost",
+        required=True, static=True)
+
+    vhost_template = ConfigText(
+        "Path to the template file to use for the vhost config",
+        default='%s/vhost.template' % (DIRNAME,), static=True)
+
+    location_template = ConfigText(
+        "Path to the template file to use for each channel's location config",
+        default='%s/location.template' % (DIRNAME,), static=True)
```

Comment:
I wonder if we should be using `pkg_resources.resource_filename` here instead of `DIRNAME`? (there are situations in which a `__file__` may not point to a real file (e.g. it may be inside a zip import), but perhaps that isn't important here).